### PR TITLE
Navigation improvements

### PR DESCRIPTION
**Pull request by gewarren**
- Add navigation helper article for the /dotnet/navigate/ URL segment - [preview](https://review.learn.microsoft.com/en-us/dotnet/navigate/?branch=pr-en-us-33716)
- Improve .NET hub page to clearly show new feature areas/TOCs - [preview](https://review.learn.microsoft.com/en-us/dotnet/?branch=pr-en-us-33716#-net--free--cross-platform--open-source-)
- Rename "toc" folder to "navigate"

### CODE REVIEW

File: docs/navigate/index.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+---
+# Navigate .NET docs
+
+Use of the following mechanisms to find other .NET documentation that you might be interested in:
+
+- The drop-down menus on the header.
+
+  <!--Add header image here once the new drop-down is added.-->
+
+- The breadcrumb trail. From any article, click on the ".NET" breadcrumb and then find another area that interests you from the [.NET documentation](../index.yml) page you're taken to.
```
Comment (BillWagner) [inline docs/navigate/index.md:16]: There are section that have move levels in the breadcrumb. Maybe make this recommendation more general:

```suggestion
- The breadcrumb trail. From any article, click on an entry in the breadcrumb trail. The ".NET" breadcrumb navigates to the [.NET documentation](../index.yml) page where you can find another area that interests you. Other entries take you to an overview for the current entry.
```

**Review by BillWagner — APPROVED** [review]
This is a great addition @gewarren 

It will help our customers learn how to navigate the content.

One other suggestion: Is it worth demonstrating how to use the "Find" function for the current TOC? I'm not sure.

File: docs/navigate/index.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+title: Navigate .NET docs
+titleSuffix: ""
+description: Learn how to navigate the .NET docs
+ms.topic: reference
+ms.date: 01/25/2023
+---
+# Navigate .NET docs
+
+Use of the following mechanisms to find other .NET documentation that you might be interested in:
```
Comment (tdykstra) [inline docs/navigate/index.md:10]:
```suggestion
Use the following mechanisms to find other .NET documentation that you might be interested in:
```

File: docs/navigate/index.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+title: Navigate .NET docs
+titleSuffix: ""
+description: Learn how to navigate the .NET docs
+ms.topic: reference
+ms.date: 01/25/2023
+---
+# Navigate .NET docs
```
Comment (tdykstra) [inline docs/navigate/index.md:8]: You could link to this page from an entry at the end of each toc.

**Review by tdykstra — APPROVED** [review]
Looks good!

File: docs/navigate/index.md (hunk 3 above)
Comment (gewarren) [reply docs/navigate/index.md:8]: I'm not crazy about adding our own custom navigation affordances to .NET docs. @dotnet/docs What are your thoughts about this?

Comment (gewarren) [reply docs/navigate/index.md:8]: Resolving this for now. If you feel strongly about adding the TOC entry, can you open an issue for it?

Comment (tdykstra) [reply docs/navigate/index.md:8]: How will people find this page otherwise?

Comment (gewarren) [reply docs/navigate/index.md:8]: The idea behind having this page was if someone hacked the URL to just /dotnet/navigate/.

If users don't know to use the L2 header or breadcrumb, then that's a much bigger problem with the Learn platform IA. This page was just intended to provide hints for the URL hack, but I think it's wrong to lean on it as the main navigation learning resource.

**Pull request merged**